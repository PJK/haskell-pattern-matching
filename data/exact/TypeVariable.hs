module TypeVariable where

data MyTrinary = A | B | C
data Container a = ContainerC a | Nope

unpack :: Container a -> a
unpack (ContainerC x) = x
unpack (ContainerC x) = x