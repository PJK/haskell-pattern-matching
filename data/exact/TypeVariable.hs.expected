{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "unpack",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        }
                    ]
                ]
            }
        ],
        [
            "unpack",
            {
                "tag": "InaccessibleRhs",
                "contents": [
                    {
                        "tag": "ConstructorPattern",
                        "contents": [
                            "ContainerC",
                            [
                                {
                                    "tag": "VariablePattern",
                                    "contents": "x"
                                }
                            ]
                        ]
                    }
                ]
            }
        ]
    ]
}